### DISCUSSION TITLE
arrays

### DISCUSSION THREAD

**Discussion by xoma-star** [pocketbase/pocketbase#539] [Q&A] [answered]
is there a way to add array type filed to my collection?
e.g. [1,2,3,4]

**Comment by ganigeorgiev** [comment, marked as answer]
You have 3 options depending on whether the values of the array are "fixed" (aka. from a known set) or arbitrary:

1. If you know all possible values that your array could have, you can create a "Mutiple choices" field. This also comes with validations and will disallow/strip any value that is not defined in your options list.

2. Create a regular `json` type field. This could be any json value - null, object, array, number, string, etc. No additional data validations are performed.

3. If your array values are complex, you could store them in a new collection and create a link to this via the `relation` type field. This will require 2-step submission (_in the future we may add "repeatable/group" type of field that will allow submission in a single transactional request_).
